File: src/python/finished/count_square_sum_triples.py
import math


# https://leetcode.com/problems/count-square-sum-triples/


class Solution:
    def countTriples(self, n: int) -> int:
        cnt = 0
        for i in range(1, n + 1):
            for j in range(1, i):
                asq = i ** 2
                bsq = j ** 2
                csq = asq + bsq
                c = math.floor(math.sqrt(csq))
                if c <= n and c ** 2 == csq:
                    cnt += 2
        return cnt

File: src/rust/finished/count_square_sum_triples.rs
impl Solution {
    pub fn count_triples(n: i32) -> i32 {
        let mut count = 0;

        for i in (1..=n) {
            for j in (1..i) {
                let asq = i.pow(2);
                let bsq = j.pow(2);
                let csq = asq + bsq;
                let c = (csq as f64).sqrt().floor() as i32;
                if c <= n && c.pow(2) == csq {
                    count += 2;
                }
            }
        }

        count
    }
}
